cli/azd/cmd/actions/action.go

```diff
@@ -32,6 +32,11 @@ type Action interface {
 	Run(ctx context.Context) (*ActionResult, error)
 }
 
+type ActionOptions struct {
+	Name             string
+	DisableTelemetry bool
```

Comment:
We should avoid exposing `DisableTelemetry` as an option if possible. I think we could accomplish this by having the telemetry middleware construction accept the flag at middleware build time.

Reply:
I removed the `ActionOptions` from the middleware interface and only those middleware needed this options can inject them in with wire.  Hopefully this will be the best of both worlds.  Still needed to export the property name though since the middlewares are in a different package.